resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/ResourceStagingServer.scala

```diff
@@ -98,8 +100,36 @@ object ResourceStagingServer {
     } else {
       new SparkConf(true)
     }
+    val apiServerUri = sparkConf.get(RESOURCE_STAGING_SERVER_API_SERVER_URL)
+    val initialAccessExpirationMs = sparkConf.get(
+        RESOURCE_STAGING_SERVER_INITIAL_ACCESS_EXPIRATION_TIMEOUT)
+    val resourceCleanupIntervalMs = sparkConf.get(RESOURCE_STAGING_SERVER_CLEANUP_INTERVAL)
     val dependenciesRootDir = Utils.createTempDir(namePrefix = "local-application-dependencies")
-    val serviceInstance = new ResourceStagingServiceImpl(dependenciesRootDir)
+    val useServiceAccountCredentials = sparkConf.get(
+        RESOURCE_STAGING_SERVER_USE_SERVICE_ACCOUNT_CREDENTIALS)
+    // Namespace doesn't matter because we list resources from various namespaces
+    val kubernetesClient = SparkKubernetesClientFactory.createKubernetesClient(
+        apiServerUri,
+        "default",
```

Comment:
We should be able to create a non-namespaced client for use here. I made a change in https://github.com/apache-spark-on-k8s/spark/pull/319 to facilitate that.